### PR TITLE
Add entrys to steps list for uncaught exceptions

### LINKED ISSUE #213: Error block content is not displayed
### Jenkins and plugins versions report

<details>
  <summary>Environment</summary>

  
  ```text
OS: Linux - 5.10.147+
Java: 11.0.18 - Eclipse Adoptium (OpenJDK 64-Bit Server VM)
---
ansicolor:1.0.2
antisamy-markup-formatter:155.v795fb_8702324
apache-httpcomponents-client-4-api:4.5.13-138.v4e7d9a_7b_a_e61
authentication-tokens:1.4
authorize-project:1.4.0
basic-branch-build-strategies:71.vc1421f89888e
blueocean:1.25.8
blueocean-autofavorite:1.2.5
blueocean-bitbucket-pipeline:1.25.8
blueocean-commons:1.25.8
blueocean-config:1.25.8
blueocean-core-js:1.25.8
blueocean-dashboard:1.25.8
blueocean-display-url:2.4.1
blueocean-events:1.25.8
blueocean-git-pipeline:1.25.8
blueocean-github-pipeline:1.25.8
blueocean-i18n:1.25.8
blueocean-jwt:1.25.8
blueocean-personalization:1.25.8
blueocean-pipeline-api-impl:1.25.8
blueocean-pipeline-editor:1.25.8
blueocean-pipeline-scm-api:1.25.8
blueocean-rest:1.25.8
blueocean-rest-impl:1.25.8
blueocean-web:1.25.8
bootstrap4-api:4.6.0-5
bootstrap5-api:5.2.1-3
bouncycastle-api:2.26
branch-api:2.1051.v9985666b_f6cc
build-with-parameters:1.6
caffeine-api:2.9.3-65.v6a_47d0f4d1fe
checks-api:1.8.1
cloudbees-bitbucket-branch-source:791.vb_eea_a_476405b
cloudbees-folder:6.795.v3e23d3c6f194
command-launcher:90.v669d7ccb_7c31
commons-lang3-api:3.12.0-36.vd97de6465d5b_
commons-text-api:1.10.0-27.vb_fa_3896786a_7
conditional-buildstep:1.4.2
configuration-as-code:1569.vb_72405b_80249
credentials:1214.v1de940103927
credentials-binding:523.vd859a_4b_122e6
display-url-api:2.3.7
docker-build-publish:1.4.0
docker-build-step:2.9
docker-commons:1.21
docker-java-api:3.2.13-68.va_875df25a_b_45
docker-workflow:528.v7c193a_0b_e67c
durable-task:501.ve5d4fc08b0be
echarts-api:5.4.0-1
envinject:2.881.v37c62073ff97
envinject-api:1.199.v3ce31253ed13
favorite:2.4.1
font-awesome-api:6.2.0-3
gcp-secrets-manager-credentials-provider:0.3.1
ghprb:1.42.2
git:4.13.0
git-client:3.13.0
git-parameter:0.9.18
git-server:99.va_0826a_b_cdfa_d
github:1.36.0
github-api:1.303-400.v35c2d8258028
github-branch-source:1696.v3a_7603564d04
github-scm-filter-aged-refs:0.2.0
handy-uri-templates-2-api:2.1.8-22.v77d5b_75e6953
htmlpublisher:1.31
instance-identity:116.vf8f487400980
ionicons-api:31.v4757b_6987003
jackson2-api:2.13.4.20221013-295.v8e29ea_354141
jakarta-activation-api:2.0.1-2
jakarta-mail-api:2.0.1-2
javadoc:226.v71211feb_e7e9
javax-activation-api:1.2.0-5
javax-mail-api:1.6.2-8
jaxb:2.3.7-1
jdk-tool:63.v62d2fd4b_4793
jenkins-design-language:1.25.8
jjwt-api:0.11.5-77.v646c772fddb_0
job-dsl:1.81
jquery:1.12.4-1
jquery3-api:3.6.1-2
jsch:0.1.55.61.va_e9ee26616e7
junit:1177.v90374a_ef4d09
kubernetes:3734.v562b_b_a_627ea_c
kubernetes-client-api:5.12.2-193.v26a_6078f65a_9
kubernetes-credentials:0.9.0
lockable-resources:2.18
mailer:438.v02c7f0a_12fa_4
matrix-auth:3.1.5
matrix-combinations-parameter:1.3.2
matrix-project:785.v06b_7f47b_c631
maven-plugin:3.20
metrics:4.2.10-389.v93143621b_050
mina-sshd-api-common:2.9.1-44.v476733c11f82
mina-sshd-api-core:2.9.1-44.v476733c11f82
nodelabelparameter:1.11.0
okhttp-api:4.9.3-108.v0feda04578cf
parameterized-trigger:2.45
pipeline-build-step:2.18.1
pipeline-graph-analysis:202.va_d268e64deb_3
pipeline-graph-view:144.vf3924feb_7e35
pipeline-groovy-lib:629.vb_5627b_ee2104
pipeline-input-step:466.v6d0a_5df34f81
pipeline-milestone-step:101.vd572fef9d926
pipeline-model-api:2.2118.v31fd5b_9944b_5
pipeline-model-definition:2.2118.v31fd5b_9944b_5
pipeline-model-extensions:2.2118.v31fd5b_9944b_5
pipeline-rest-api:2.31
pipeline-stage-step:296.v5f6908f017a_5
pipeline-stage-tags-metadata:2.2118.v31fd5b_9944b_5
pipeline-stage-view:2.31
plain-credentials:139.ved2b_9cf7587b
plugin-usage-plugin:4.0
plugin-util-api:2.20.0
popper-api:1.16.1-3
popper2-api:2.11.6-2
prometheus:2.1.1
pubsub-light:1.17
rebuild:1.34
role-strategy:562.v44e9a_e828d0e
run-condition:1.5
saml:4.372.v89f13e4c9e97
scm-api:631.v9143df5b_e4a_a
script-security:1229.v4880b_b_e905a_6
snakeyaml-api:1.33-90.v80dcb_3814d35
sse-gateway:1.26
ssh-agent:295.v9ca_a_1c7cc3a_a_
ssh-credentials:305.v8f4381501156
sshd:3.249.v2dc2ea_416e33
structs:324.va_f5d6774f3a_d
timestamper:1.21
token-macro:308.v4f2b_ed62b_b_16
trilead-api:2.72.v2a_3236754f73
uno-choice:2.6.4
variant:59.vf075fe829ccb
workflow-aggregator:590.v6a_d052e5a_a_b_5
workflow-api:1208.v0cc7c6e0da_9e
workflow-basic-steps:980.v82219a_ed188e
workflow-cps:3618.v13db_a_21f0fcf
workflow-durable-task-step:1210.va_1e5d77e122b
workflow-job:1268.v6eb_e2ee1a_85a
workflow-multibranch:716.vc692a_e52371b_
workflow-scm-step:400.v6b_89a_1317c9a_
workflow-step-api:639.v6eca_cd8c04a_a_
workflow-support:839.v35e2736cfd5c
  ```

</details>

### What Operating System are you using (both controller, and any agents involved in the problem)?

Linux

### Reproduction steps

I am using error block that contains 50 line of the output from internal task in case of failure. This content is properly displayed as output log with blue ocean but is not working (showing anything) with this plugin.

### Expected Results

I expect that output log should contains content of error block as in blue ocean

Screenshot 2023-02-20 at 01 03 34

### Actual Results

Screenshot 2023-02-20 at 00 56 15

### Anything else?

Actual content was reduced from screan but contains expected content.

### PR DESCRIPTION
**Pull request by timbrown5**
Some failed steps were getting missing.
Check for ErrorAction in nodes and if present add the node. This will contain some error text and (possibly) an exception.

This is enough for actions that return an AbortException, and to fix #213, but wouldn't include errors like: https://github.com/jenkinsci/pipeline-graph-view-plugin/issues/213#issuecomment-1438666811

To do that we:
Check if the last node (the first one we come to) has an ErrorAction - if so there is an uncaught exception, then:
- Find and record the originating step.
- Check each atomNode and endNode as we go to see if it's the originating node.
-- Once we find the originating step we add that to the `stageSteps` list - it will be mapped to the stage when we hit the start node.

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira

### CODE REVIEW

**Comment by timbrown5** [comment]
Manual Tests:
```
stage('failure') {
    echo "This will work"
    echo "THis won't - calling an undefined variable ${undefined}"
}
```
Screenshot:
Screenshot 2023-02-22 at 09 47 34

Pipeline from https://github.com/jenkinsci/pipeline-graph-view-plugin/issues/213#issuecomment-1438322664

Screenshot:
Screenshot 2023-02-22 at 20 52 58

File: src/main/java/io/jenkins/plugins/pipelinegraphview/consoleview/PipelineConsoleViewAction.java [hunk 1]
```diff
@@ -103,46 +101,59 @@ public HttpResponse getConsolOutput(StaplerRequest req) throws IOException {
...
+          if (startByte < 0) {
+            logger.info(
+                "consoleJson - requested negative startByte '"
+                    + Long.toString(startByte)
+                    + "' out of bounds, setting to 0.");
+            startByte = 0L;
+          }
         }
+
+        logger.info(
```
Comment (timja) [inline src/main/java/io/jenkins/plugins/pipelinegraphview/consoleview/PipelineConsoleViewAction.java:146]: shouldn't this be debug?

File: src/main/java/io/jenkins/plugins/pipelinegraphview/consoleview/PipelineConsoleViewAction.java [hunk 2]
```diff
@@ -103,46 +101,59 @@ public HttpResponse getConsolOutput(StaplerRequest req) throws IOException {
...
+            "Returning '"
+                + Long.toString(textLength - startByte)
+                + "' bytes from 'getConsolOutput'.");
```
Comment (timja) [inline src/main/java/io/jenkins/plugins/pipelinegraphview/consoleview/PipelineConsoleViewAction.java:149]:
```suggestion
                + "' bytes from 'getConsoleOutput'.");
```

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
spotless is failing

**Comment by timbrown5** [comment]
> spotless is failing

I've been fighting spotbugs and spotless. Not to self, `mvn clean test spotless:apply spotbugs:check` seems to be working better than running them individually.

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
> > spotless is failing
> 
> I've been fighting spotbugs and spotless. Not to self, `mvn clean test spotless:apply spotbugs:check` seems to be working better than running them individually.

Weird, I've always just run:
```
mvn spotless:apply
```
and never had an issue with it

**Comment by timbrown5** [comment]
Added two small changes to the look of the console log cards:
Fixed the alignment with the stage details card.
Screenshot 2023-02-23 at 10 07 38

Changed the expand icon color back:
Screenshot 2023-02-23 at 10 16 02
(I broke this in https://github.com/jenkinsci/pipeline-graph-view-plugin/commit/61fcf9a7fbe7d36af1ace87c35d8ce0a5c5c12c5 - turns out it wasn't unused.

**Comment by timbrown5** [comment]
Also fixed the text-overflow in card headers:
Before:
Screenshot 2023-02-23 at 11 20 05

After:
Screenshot 2023-02-23 at 11 18 35

(I was a little too eager with my cleanup before).

**Comment by timbrown5** [comment]
I think the code I added to fix spotbugs issues failed spotless, running
both means I don't miss something 😅

**Pull request merged**